### PR TITLE
Docker setup init

### PR DESCRIPTION
**Pull request by drvcodenta**
# How to Set Up the Project
### Prerequisites
Ensure the following are installed on your system before proceeding:

- Docker
- Docker Compose
- PostgreSQL client (optional, for manual database interaction)

# Step-by-Step Setup
### 1. Ensure LF Line Endings
Before building the Docker images, ensure all files, especially scripts like entrypoint.sh, .env, docker-compose.yml, Dockerfile, settings.py use LF line endings. Using CRLF can cause build failures. To verify and correct line endings:

1. If you're working on a Windows machine or collaborating across different operating systems, ensure consistent line endings:
- Set `core.autocrlf=input` in Git configurations to enforce LF-style line endings in the repository while preserving your local OS line endings.
`git config --global core.autocrlf input`
- Alternatively, in VS Code, you can manually change the line endings:
- Open the file in the editor.
- Look for the line ending type displayed in the bottom-right corner of the VS Code window (e.g., CRLF or LF).
- Click it and select "LF: Unix" from the dropdown to switch the line endings to LF.

2. For **local development**, make the following adjustments in `/blt/settings.py` to enable access over **HTTP**:
- Set `SECURE_SSL_REDIRECT=False`
- Comment out the line:
 `SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")`
- #### Convert to LF (if needed):
##### Using dos2unix
`dos2unix entrypoint.sh`
##### ⚠️ Important:
-  If line endings are not set to LF, running docker-compose build may fail.
- Avoid creating a PR to commit these local changes back to the repository.

### 2. PostgreSQL Setup
The PostgreSQL database listens on a port specified in the .env file.
Default: 5432
Example: If you encounter conflicts, it might be set to another port (e.g., 5433 in some cases). Adjust the .env file accordingly.
### 3. Application Redirections
The Django application in this setup operates over HTTP (not HTTPS).
For local development, the necessary redirects in settings.py are configured to use HTTP instead of HTTPS.
### 4. Containerized Environment
The environment includes both the application and the database as containerized services.
A bridge network facilitates communication between these containers.
### 5. Dependencies
All essential dependencies are installed and the project is functional.
### 6. Execution Details
When you run docker-compose up:

- Migrations: It checks and applies database migrations as needed.
- Static Files: Collects static files if required.
- Server: Launches the development server on http://localhost:8000.
User credentials from the .env file are automatically supplied into the containers during this process.
### 7. Configuration
A .env file is required to store PostgreSQL credentials and other essential settings:
Required Fields:
- POSTGRES_USER
- POSTGRES_PASSWORD
- POSTGRES_DB
The .env file also specifies port mappings, app-to-DB interactions, and credential configurations.
#### ⚠️ Important
If .env is omitted, you must manually adjust configurations by removing both .env and .env.example. This alternative setup is not recommended.
### 8. Inter-Container Communication
- #### Application:
1. Connects to the database using credentials and port information from the .env file.
2. Default Database Port: 5432.
- #### Database:
1. Includes a health check to verify readiness.
2. Named postgres-db in the container environment.
4. Listens on 5432 (or the port specified in .env).
### Sentry DSN Issue
- If the Sentry DSN is commented out in the environment configuration, the container repeatedly attempts to connect, causing eventual failures.
#### Solution:
- #### Option 1: Leave the Sentry configuration commented out in settings.py if you don't intend to use Sentry.
- #### Option 2: Provide your personal Sentry DSN URL in the .env file and ensure it is properly referenced in settings.py.
# Commands to Set Up the Project
- #### First check the LF Line Endings defined above

- #### Copy and configure the .env file:
`cp .env.example .env`
Update credentials and settings as needed.

- #### Build the Docker images:

`docker-compose build`
- #### Start the containers:

`docker-compose up`
- #### Access the application:

- Open your browser and navigate to:
http://localhost:8000/
- #### Prevent Automatic Redirects to HTTPS: 
- Modern browsers often redirect HTTP requests to HTTPS if a previous secure connection was established. To prevent this:
- Use Incognito Mode (Private Browsing): Open the browser in incognito mode and access the application using http://localhost:8000.
- Clear HSTS Settings (if needed): 
If redirection persists, clear your browser's HSTS settings:
Chrome: Go to chrome://net-internals/#hsts, and under "Delete domain security policies," enter localhost and click "Delete."
- Firefox: Clear the site's history to reset HSTS settings.
- Ensure you're explicitly using http:// instead of https:// in the URL.
### Notes
- The project listens on port 8000 over the HTTP protocol.
- Ensure all required configurations in .env are correct for seamless setup.

### Error Edge Cases
- If container fails execute ./entrypoint.sh due to permission error, use `chmod +x ./entrypoint.sh`
- If you encounter ./entrypoint.sh was not found, then make sure you are using `LF` line ending in place of `CRLF`
- If you encounter ERR_SSL_PROTOCOL_ERROR when you try to access the server on http://localhost:8000, make sure the Browser doesn't automatically redirect to https://localhost:8000. If it keeps doing this behaviour, then you can set `SECURE_SSL_REDIRECT` to `False` locally only(search for it  /blt/settings.py), stop the container and start it again.
- If you encounter the same error indicating SSL_REDIRECT in the logs while building the container, set `SECURE_SSL_REDIRECT` to `False`

### CODE REVIEW

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
A few changes requested

**Review by DonnieBLT — CHANGES_REQUESTED** [review]
one change

File: blt/settings.py [hunk 1]
```diff
@@ -310,7 +310,7 @@
 
 # Honor the 'X-Forwarded-Proto' header for request.is_secure()
 
-SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
+# SECURE_PROXY_SSL_HEADER = ("HTTP_X_FORWARDED_PROTO", "https")
```
[the remark this answers is not in the document]
Comment (drvcodenta) [reply blt/settings.py:313]: ok

**Comment by drvcodenta** [comment]
#2801
#2808

**Review by DonnieBLT — DISMISSED** [review]

**Review by DonnieBLT — APPROVED** [review]

**Pull request merged**